c10/core/SymInt.cpp

```diff
@@ -68,13 +93,11 @@ bool SymInt::operator<(int64_t sci) const {
 }
 
 bool SymInt::operator==(int64_t sci) const {
-  TORCH_CHECK(!this->is_symbolic(), "Symbolic eq isn't supported yet");
-  return data_ == sci;
+  return *this == c10::SymInt(sci);
 }
 
 bool SymInt::operator!=(int64_t sci) const {
-  TORCH_CHECK(!this->is_symbolic(), "Symbolic neq isn't supported yet");
-  return data_ != sci;
+  return !(*this == sci);
```

Comment:
This is worse, because we don't have symbolic bools, so you'll end up guards going the wrong direction. Need to bind not equals separately